server/handlers.go

```diff
@@ -141,18 +142,33 @@ func (ah *ActivityHandler) Handle(activity msteams.Activity) error {
 }
 
 func (ah *ActivityHandler) HandleLifecycleEvent(event msteams.Activity) {
-	if event.LifecycleEvent == "reauthorizationRequired" {
-		expiresOn, err := ah.plugin.GetClientForApp().RefreshSubscription(event.SubscriptionID)
-		if err != nil {
-			ah.plugin.GetAPI().LogWarn("Unable to refresh the subscription", "error", err.Error())
-			ah.plugin.GetMetrics().ObserveLifecycleEvent(event.LifecycleEvent, metrics.DiscardedReasonFailedToRefresh)
-			return
-		}
+	if event.LifecycleEvent != "reauthorizationRequired" {
+		ah.plugin.GetMetrics().ObserveLifecycleEvent(event.LifecycleEvent, metrics.DiscardedReasonUnknownLifecycleEvent)
+		return
+	}
 
-		ah.plugin.GetMetrics().ObserveSubscription(metrics.SubscriptionRefreshed)
-		if err = ah.plugin.GetStore().UpdateSubscriptionExpiresOn(event.SubscriptionID, *expiresOn); err != nil {
-			ah.plugin.GetAPI().LogWarn("Unable to store the subscription new expiry date", "subscription_id", event.SubscriptionID, "error", err.Error())
-		}
+	// Ignore subscriptions we aren't tracking locally. For now, that's just the single global chats subscription.
```

Comment:
We would renew everything, even if unused -- lots of time on msteams-sync-test doing this until I upgraded to a tes build.